src/zarr/codecs/sharding.py

```diff
@@ -760,6 +771,115 @@ async def _load_full_shard_maybe(
             else None
         )
 
+    async def _load_partial_shard_maybe(
```

Comment:
What would happen if we got rid of `_load_full_shard_maybe` and instead treated loading the full shard as a special case of loading some of the shard?